### ISSUE TITLE
[Feature]: [Currency Input] Provide option to choose decimal separator and thousand separator characters

### ISSUE THREAD

**Issue by sbalaji1192** [appsmithorg/appsmith#15258]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Currently, the thousand separator and decimal separator characters are inferred from the locale. As part of this feature, we need to provide the option to configure the character that must be used while formatting the entered value. 

### Why should this be worked on?

This enables the internalization of the currency widget.

**Comment by keisyd** [comment]
I agree. Here in Brazil the thousands are not comma separated, so I will no longer be able to use the widget...

**Comment by RawkHogan** [comment]
When accessing the app from a mobile phone, when I add a comma in the currency field it erases what I typed before.

**Comment by glarghetti** [comment]
Even in Italy we have the same problem, I can't use the plugin until it is fixed.
Too bad I found out after investing many hours, I hope it will be fixed soon.

**Comment by GreenFlux** [comment]
Another user in Discord is facing this issue: 
https://discord.com/channels/725602949748752515/725609493974614076/1267531604658749623

I see no one is assigned to this issue, so tagging a few for visibility.
@jacquesikot @rahulbarwal @nerbos

**Comment by akhabibulin-casecreate** [comment]
up

**Comment by habermst** [comment]
Agree – that would be great. We can't have commas becuase we use ', so unfortunately the currency widget is useless.

**Comment by rahulbarwal** [comment]
Thank you folks for the interest and requests for this feature. This is not a priority at the moment. We will  get back to this later.

**Comment by hugorn** [comment]
In the table widget I have the same problem, my decimal separator is a comma but the number type field does not accept typing this character.

**Comment by tomjose92** [comment]
There was an issue where the currency widget was not using comma as decimal separator, based on the locale.
This issue has been fixed for all currency related fields like Currency widget and in Currency column in the Table widget.

@glarghetti, @heberb, @keisyd, @akhabibulin-casecreate If you need to see comma as decimal separator, use the currency widget and ensure the default locale of the browser is something that supports this like Brazil, Portuguse or other European locales.

@hugorn , this github issue was created for currency widget. Can you create another issue requesting for support of this functionality in number input widget or number column

Closing this ticket
